src/htmlize/patchtml.ml

```diff
@@ -20,10 +19,115 @@ let edit_html ?header ?footer ~head_childs html =
     footer
     "#footer";
   let head = html $ "head" in
-  List.iter (fun (id,child) -> 
-      opt_inserter
-        (iter (fun e -> append_child head e))
-        (Some child)
-        id)
+  List.iter (fun (id,child) ->
+    opt_inserter
+      (iter (fun e -> append_child head e))
+      (Some child)
+      id)
     head_childs;
   to_string html
+
+open Omd
+(* Keeps us from processing url links *)
+let is_url url =
+  (String.length url >= 7 && String.equal (String.sub url 0 7) "http://")
+  || (String.length url >= 8 && String.equal (String.sub url 0 8) "https://")
+
+(* Keeps us from processing anchors *)
+let is_anchor link =
+  String.length link >= 1 && String.equal (String.sub link 0 1) "#"
+
+(* Establish depth of given link relative to the root
+ * Then prepend the corresponding number of .. to the basename
+ * Then append index.html to fix link
+ * *)
+let handle_link link =
+  if link.destination = "" then
+    link
+
+  (* Escape through this branch if the link points to an http(s) address,
+   * or is a link from a Table of Contents
+   * *)
+  else if is_url link.destination || is_anchor link.destination then
+    link
+
+  else
+    (* Turn the link into a list in order to process it *)
+    let li = String.split_on_char '/' link.destination in
```

Comment:
What if a link contains `\/` e.g. `hello\/ooo/` ?

Reply:
I believe "\" characters is an illegal filename character, isn't it ? It can be present in a link when manually typed, but never in a valid filename, therefore making the link point to an impossibly named file ? Am I mistaken ? What's your opinion on the matter ? Should we sanitize the link from illegal filename characters before processing it ?

Reply:
Indeed, my bad.